cmd/limactl/shell.go

```diff
@@ -128,9 +128,21 @@ func shellAction(cmd *cobra.Command, args []string) error {
 	}
 	script := fmt.Sprintf("%s ; exec %s --login", changeDirCmd, shell)
 	if len(args) > 1 {
+		quotedArgs := make([]string, len(args[1:]))
+		parsingEnv := true
+		for i, arg := range args[1:] {
+			var quotedArg string
+			if parsingEnv && isEnv(arg) {
+				quotedArg = quoteEnv(arg)
+			} else {
+				parsingEnv = false
+				quotedArg = shellescape.Quote(arg)
+			}
+			quotedArgs[i] = quotedArg
```

Comment:
This is nitpicking now, but my preferred way to write it would be:

```suggestion
			if parsingEnv && isEnv(arg) {
				quotedArgs[i] = quoteEnv(arg)
			} else {
				quotedArgs[i] = shellescape.Quote(arg)
				parsingEnv = false
			}
```

Reply:
Yeah, this looks a lot neater to me too. Updated